### PR TITLE
fix(TDI-38338): add boolean type to filter expressions table

### PR DESCRIPTION
**Pull request by akhabali**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
please refer to : https://jira.talendforge.org/browse/TDI-38338

**What is the new behavior?**
the boolean type is available in the filter expression table and can be used to construct filters

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/9/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/9/)

**Comment by akhabali** [comment]
@sgandon there are tests for the filter expression generation. 
i added some to cover the new boolean type filter.
The filter expression is generated using Microsoft azure api, so i think there is no need to have integration tests for this. unit tests are sufficient.

File: components/components-azurestorage/src/test/java/org/talend/components/azurestorage/table/helpers/FilterExpressionTableTest.java [hunk 1]
```diff
@@ -116,6 +134,18 @@ public void testFilterExpressionTable() {
...
         assertEquals(query,
                 "(((PartitionKey eq '12345') and (RowKey gt 'AVID12345')) or (Timestamp ge datetime'2016-01-01 00:00:00')) or (AnUnknownProperty lt 'WEB345')");
+
+        // Boolean
+        clearLists();
+        columns.add("ABooleanProperty");
+        functions.add(COMPARISON_EQUAL);
+        values.add("true");
+        predicates.add(PREDICATE_AND);
+        fieldTypes.add(FIELD_TYPE_STRING);
```
Comment (gonchar-ivan) [inline components/components-azurestorage/src/test/java/org/talend/components/azurestorage/table/helpers/FilterExpressionTableTest.java]: A little bit strange for me
Maybe here should FIELD_TYPE_BOOLEAN?

Comment (akhabali) [inline components/components-azurestorage/src/test/java/org/talend/components/azurestorage/table/helpers/FilterExpressionTableTest.java]: That's need to be boolean. It's a little mistake

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/14/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/14/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/21/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/21/)

**Pull request merged**